File: JavaCodes/UnionAndIntersectionOfTwoArrays.java
import java.util.*;

public class UnionAndIntersectionOfTwoArrays {
    public static void main(String[] args) {
        int m, n;
        System.out.println("Enter the numner of elements of the 1st and 2nd arrays respectively");
        Scanner sc = new Scanner(System.in);
        m = sc.nextInt();
        n = sc.nextInt();
        System.out.println("Enter the elements of the 1st array");
        Integer arr[] = new Integer[m];
        for (int i = 0; i < m; i++)
            arr[i] = sc.nextInt();
        System.out.println("Enter the elements of the 2st array");
        Integer brr[] = new Integer[n];

        for (int j = 0; j < n; j++)
            brr[j] = sc.nextInt();
        System.out.println("Union of the two arrays ");

        Set<Integer> un = new HashSet<>();
        for (int i = 0; i < m; i++)
            un.add(arr[i]);
        for (int j = 0; j < n; j++)
            un.add(brr[j]);

        for (int ele : un)
            System.out.print(ele + " ");

        System.out.println("\nThe common elements are ");
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (arr[i] == brr[j])
                    System.out.print(arr[i] + " ");
            }
        }

    }
}

// java has "Set" data structure that stores only distinct elements.
// the syntax is --> Set<data_type> un = new HashSet<>();

File: cppCode/UnionAndIntersectionOfTwoArrays.cpp
#include <iostream>
#include<iterator>
#include<set>
using namespace std;
int main() {
        int m, n;
        cout<<"Enter the numner of elements of the 1st and 2nd arrays respectively";
        
        cin>>m;
        cin>>n;
        cout<<"Enter the elements of the 1st array";
        int arr[m];
        for (int i = 0; i < m; i++)
            cin>>arr[i];
        cout<<"Enter the elements of the 2st array";
        int brr[n];

        for (int j = 0; j < n; j++)
            cin>>brr[j];
        cout<<"Union of the two arrays ";
        
        set<int, greater<int>> un;
        for (int i = 0; i < m; i++)
            un.insert(arr[i]);

            for (int j = 0; j < n; j++)
            un.insert(brr[j]);

        for (int ele : un)
            cout<<ele<<" ";

        cout<<"\nThe common elements are ";
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (arr[i] == brr[j])
                    cout<<arr[i]<<" ";
            }
        }

    }

// c++ has "set" STL that stores only distinct elements.
// the syntax is --> set<datatype> setname;
//  STL belongs inside #include<set> header file. #include<iterator> file should also be called.